src/vsr/replica.zig

```diff
@@ -835,7 +835,12 @@ pub fn ReplicaType(
             const prepare_timestamp = self.state_machine.prepare(
                 message.header.operation.cast(StateMachine),
                 message.body(),
-            );
+            ) catch |err| switch (err) {
+                error.NonZeroTimestamp => {
+                    log.err("{}: on_request: dropping (non zero timestamp)", .{self.replica});
```

Comment:
Additionally log the `message.header.client` and the `message.header.operation.cast(StateMachine)` — they could help pinpoint the origin.